Hi

TheFrenchLeaf opened the pull request:
Hi Engin,
It's Pierre (pmoulon@gmail.com)

I purpose you some code change in libDaisy

First a code simplification by avoiding code duplicates for the convolution

Add CMakeLists build rules to compile daisy as a library easily through CMake.

It could be fun now to see if we could put some Openmp things to make daisy multicore able ;-)

Regards,
Pierre

Hi hope everything is fine with your new projects.

pmoulon committed:
Change convolution to be a template function (it's more generic). It avoids code duplicates.
Add CMakeLists build rules to compile daisy as a library.

etola commented:
did you make any speed experiments after changing the convolution code ? I don't like to have 'too generic' code for optimized operations. daisy depends hugely on convolutions and I expect them to be float since the descriptor is computed in float. For example I do vertical convolutions for multiple columns in a go to avoid cache misses and if someone passes double along the way then those optimized convolution operations will slow down...

also, for your multi core remarks... there are already openmp preprocessor instructions in the code. if it doesn't work for you, you need to enable them in the makefile.

I'll check cmakelist and incorporate it.

pmoulon committed:
Fix warnings.

pmoulon committed:
Fix compilation on windows.

pmoulon committed:
Add openmp support.
Add compilation of the main.cpp file. Support only grayscale ppm files natively.

TheFrenchLeaf commented:
Hi Engin,

I have fix the OpenMP auto detection in the Cmake compilation rules and add the compilation of the main.cpp file.

The compiler do the job to create the good function in regards of the template instantiation in the code.
You could check if performance change occur for you on linux. On windows I do not observe variation in the timings.

Regards.

Later I could manage the jpeg and png detection and linking under linux withing the cmakeFile.

pmoulon committed:
Fix the definition of a preprocessor in the OpenMP compilation.
